protoc_polyglot/cli.py

```diff
@@ -0,0 +1,181 @@
+    
+class Tools:
+    @staticmethod
+    def get_service_info(services_yaml, name:str, key:str = 'files') -> dict[str, list[str]] | list[str]:
+        with open(services_yaml, 'r') as file:
+            data = yaml.safe_load(file)
+            
+            if name=="":
+                return {n: data[n][key] for n in data}
+            else:
+                return data[name][key]
+
+    @staticmethod
+    def get_files_from_directory(dir: str):
+        proto_files = []
+        for root, directories, files in os.walk(dir):
+            for filename in files:
+                if ".proto" in filename:
+                    proto_files.append(join(root, filename))
+        return proto_files
+
+    @staticmethod
+    def zip_directory(dir: str):
+        zip_file = f'{dir}.zip'
+        with ZipFile(zip_file, 'w') as zip:
+            for root, directories, files in os.walk(dir):
+                for filename in files:
+                    filepath = join(root, filename)
+                    zip.write(filepath, filepath[len(dir):])
+        return zip_file
+
+    @staticmethod
+    def download_file(name, saveto):
+        url = f"https://github.com/quantasoftcz/protoc-polyglot/files/{name}"
+        response = requests.get(url)
+        
+        if response.status_code == 200:
+            os.makedirs(dirname(saveto), exist_ok=True)
+            with open(saveto, 'wb') as f:
+                f.write(response.content)
+            return True
+        else:
+            print(f"Failed to download {url}. Status code: {response.status_code}")
+            return False
+
+class Base_UI:
+    def __init__(self, settings=None):
+        if settings:
+            self.settings = settings
+        else:
+            self.settings = Settings()
+
+    def get_plugin_path(self) -> str:
+        return join(self.settings.plugins_base_path, self.plugin_name)
+    
+    def get_grpc_path(self) -> str:
+        return "plugins/grpc"
+    
+    def download_grpc_and_protobuf(self):
+        name = f"15369526/grpc-{self.settings.grpc_version}_protobuf-{self.settings.protobuf_version}.zip"
+        if not os.path.exists(self.get_grpc_path()):
+            print("gRPC missing, downloading...")
+        return Tools.download_file(name, self.get_grpc_path())
+        
+    def download_plugin(self):
+        if not os.path.exists(self.get_plugin_path()):
+            print("Plugin missing, downloading...")
+        return Tools.download_file("grpc_python_plugin", self.get_plugin_path())
+            
+    
+    def list(self):
+        print(os.getcwd())
+        
+        with os.scandir(self.settings.CORE_DIR) as entries:
+            print('Supported languages:')
+            folders = [entry.name for entry in entries if entry.is_dir() and entry.name not in ['__pycache__']]
+            for f in folders:
+                print(f)
+        
+        # if self.settings.CORE_DIR == '':
+        #     print('Hello world from polyglot')
+        #     exit()
```

Comment:
Shouldn't it have a plugin name as an  argument?

Reply:
Nevim presne co myslis. Jako tahle funkce je trochu divna, to jo. Nevedel jsem totiz poradne co s ni, protoze napr. Python package se k ni vubec nedostane (podporuje jen protoc-polyglot [language] [command] ... - coz je na dalsi diskuzi)

Reply:
Ok, it's deleted